Tests/SwiftGRPCNIOTests/GRPCChannelHandlerResponseCapturingTestCase.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+import NIO
+import NIOHTTP1
+@testable import SwiftGRPCNIO
+import XCTest
+
+class CollectingChannelHandler<OutboundIn>: ChannelOutboundHandler {
+  var responses: [OutboundIn] = []
+
+  func write(ctx: ChannelHandlerContext, data: NIOAny, promise: EventLoopPromise<Void>?) {
+    responses.append(unwrapOutboundIn(data))
+  }
+}
+
+class CollectingServerErrorDelegate: ServerErrorDelegate {
+  var errors: [Error] = []
+
+  func observe(_ error: Error) {
+    self.errors.append(error)
+  }
+}
+
+class GRPCChannelHandlerResponseCapturingTestCase: XCTestCase {
+  static let echoProvider: [String: CallHandlerProvider] = ["echo.Echo": EchoProvider_NIO()]
+  class var defaultServiceProvider: [String: CallHandlerProvider] {
+    return echoProvider
+  }
+
+  func configureChannel(withHandlers handlers: [ChannelHandler]) -> EventLoopFuture<EmbeddedChannel> {
+    let channel = EmbeddedChannel()
+    return channel.pipeline.addHandlers(handlers, first: true)
+      .map { _ in channel }
+  }
+
+  var errorCollector: CollectingServerErrorDelegate = CollectingServerErrorDelegate()
+
+  override func setUp() {
+    errorCollector.errors.removeAll()
```

Comment:
Is this necessary? I thought that a separate test class instance gets instantiated for each test case.

If you do need to keep this, please make sure to call `super.setUp`, though.